packages/bpk-component-theme-switcher/src/updateOnThemeChange.js

```diff
@@ -0,0 +1,76 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { Component, PropTypes } from 'react';
+
+import { wrapDisplayName } from 'bpk-react-utils';
+
+import { getHtmlElement, THEME_CHANGE_EVENT } from './utils';
+
+const updateOnThemeChange = (EnhancedComponent) => {
+  class UpdateOnThemeChange extends Component {
+    constructor() {
+      super();
+      this.onThemeChange = this.onThemeChange.bind(this);
+      this.state = {
+        theme: null,
+      };
+    }
+
+    componentDidMount() {
+      getHtmlElement().addEventListener(THEME_CHANGE_EVENT, this.onThemeChange, false);
+    }
+
+    componentWillUnmount() {
+      getHtmlElement().removeEventListener(THEME_CHANGE_EVENT, this.onThemeChange, false);
+    }
+
+    onThemeChange(e) {
+      const { theme } = e.detail;
+      this.setState({ theme });
+      this.forceUpdate();
+    }
+
+    render() {
+      return !this.state.theme ?
+        (
+          <div>
+            {this.props.children}
+          </div>
+        )
+        :
+        (
+          <div>
+            <EnhancedComponent
```

Comment:
Wouldn't we always render the enhanced component and just let the theme prop pass through?

Reply:
If you render it without providing the theme props you get prop type warnings, which we might be okay with.

Reply:
Not with the current set up of the theme provider, if the theme is invalid will throw errors, that's why the conditional render